src/inspector/models/score/scoredisplaysettingsmodel.h

```diff
@@ -40,10 +42,19 @@ public slots:
     void shouldShowPageMarginsChanged(bool shouldShowPageMargins);
 
 private:
-    bool m_shouldShowInvisible = true;
-    bool m_shouldShowUnprintable = true;
-    bool m_shouldShowFrames = true;
-    bool m_shouldShowPageMargins = false;
+    void updateShouldShowInvisible(bool isVisible);
+    void updateShouldShowUnprintable(bool isVisible);
+    void updateShouldShowFrames(bool isVisible);
+    void updateShouldShowPageMargins(bool isVisible);
+
+    void updateFromConfig(mu::notation::ScoreConfigType configType);
+    void updateAll();
+    void setupConnections();
+
+    bool m_shouldShowInvisible = adapter()->isNotationExisting() ? adapter()->scoreConfig().isShowInvisibleElements : true;
+    bool m_shouldShowUnprintable = adapter()->isNotationExisting() ? adapter()->scoreConfig().isShowUnprintableElements : true;
```

Comment:
Better to put this code to the constructor of this class and here you should use only default values:

bool m_shouldShowInvisible = false;
bool m_shouldShowUnprintable = false;
...